### PR TITLE
[RTG][RTGTest] Add Python Bindings

### PR DESCRIPTION
**Pull request by maerhart**

### CODE REVIEW

File: lib/Bindings/Python/CMakeLists.txt [hunk 1]
```diff
@@ -9,39 +9,58 @@ add_compile_definitions("MLIR_PYTHON_PACKAGE_PREFIX=circt.")
...
+)
+
+set(PYTHON_BINDINGS_SOURCES
+  CIRCTModule.cpp
+  ESIModule.cpp
+  HWModule.cpp
+  OMModule.cpp
+  RTGModule.cpp
+  RTGToolModule.cpp
+  MSFTModule.cpp
```
Comment (uenoku) [inline lib/Bindings/Python/CMakeLists.txt:23]: nit: Please alpha-sort.

File: lib/Bindings/Python/RTGModule.cpp [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+          },
+          py::arg("self"), py::arg("ctxt") = nullptr);
+
+  mlir_type_subclass(m, "SetType", rtgTypeIsASet)
+      .def_classmethod(
+          "get",
+          [](py::object cls, MlirType elementType) {
+            return cls(rtgSetTypeGet(elementType));
+          },
+          py::arg("self"), py::arg("element_type"));
```
Comment (uenoku) [inline lib/Bindings/Python/RTGModule.cpp:41]: Could you add a test for these?

File: integration_test/Bindings/Python/dialects/rtg.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
+# REQUIRES: bindings_python
+# RUN: %PYTHON% %s | FileCheck %s
+
+import circt
+
+from circt.dialects import rtg, rtgtest
```
Comment (uenoku) [inline integration_test/Bindings/Python/dialects/rtg.py:6]: Does this test require a guard for `CIRCT_INCLUDE_TESTS` if it uses rtgtest?

File: lib/Bindings/Python/RTGModule.cpp [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+          [](py::object cls, MlirContext ctxt, py::list entryNames,
+             py::list entryTypes) {
+            std::vector<MlirAttribute> names;
+            std::vector<MlirType> types;
+            for (auto type : entryNames)
+              names.push_back(type.cast<MlirAttribute>());
+            for (auto type : entryTypes)
+              types.push_back(type.cast<MlirType>());
+            assert(names.size() == types.size() &&
+                   "number of entry names and entry types must match");
```
Comment (uenoku) [inline lib/Bindings/Python/RTGModule.cpp:55]: nit: Please consider to raise exception instead of assertion failure for this api error.

File: include/circt/Dialect/RTGTest/IR/RTGTestOps.td [hunk 5]
```diff
@@ -20,7 +20,6 @@ class RTGTestOp<string mnemonic, list<Trait> traits = []> :
 
 def CPUDeclOp : RTGTestOp<"cpu_decl", [
   Pure,
-  ConstantLike,
```
Comment (uenoku) [inline include/circt/Dialect/RTGTest/IR/RTGTestOps.td:23]: Why is this removed?

**Review by uenoku — COMMENTED** [review]
Python binding looks great to me.

**Review by uenoku — APPROVED** [review]

File: include/circt/Dialect/RTGTest/IR/RTGTestOps.td (hunk 5 above)
Comment (maerhart) [reply include/circt/Dialect/RTGTest/IR/RTGTestOps.td:23]: Operations marked with `ConstantLike` require a folder which is missing. I was wondering whether I should just add one returning the IntegerAttr or if I should add a custom attribute for that, so I took the easy way out by removing it for now and adding it later again together with the folder. But I'm also happy to add the folder right here if you prefer.

File: integration_test/Bindings/Python/dialects/rtg.py (hunk 3 above)
Comment (maerhart) [reply integration_test/Bindings/Python/dialects/rtg.py:6]: The whole `integration_test` folder is already guarded by `CIRCT_INCLUDE_TESTS`

File: include/circt/Dialect/RTGTest/IR/RTGTestOps.td (hunk 5 above)
Comment (uenoku) [reply include/circt/Dialect/RTGTest/IR/RTGTestOps.td:23]: Got, it. I was just curious so please feel free to keep as is.

**Review by fabianschuiki — APPROVED** [review]
LGTM!
